src/generators.py

```diff
@@ -0,0 +1,347 @@
+    the character is printed n times, where n is a random value between 'minstreak' and 'maxstreak'
+    (included)
+    chainLength: Expects an integer > 0 to determine the length of the string output
+    alphabet: A Python list containing all valid characters
+    prob: A probability between 0 and 1 (0.2)
+    minstreak: The minimum amount of characters that will appear in a streak (3)
+    maxstreak: The maximum amount of characters that will appear in a streak (100)
+    """
+    prob = 0.2
+    minstreak = 3
+    maxstreak = 100
+    output = ""
+    for i in range(chainLength):
+        selectedChar = random.choice(alphabet)
+        if random.random()>=prob:
+            output += selectedChar
+        else:
+            streakLength = random.choice(range(minstreak, maxstreak+1))
+            if streakLength > chainLength-i:
+                streakLength = i
+            output += "".join(selectedChar for _ in range(streakLength))
+            i += streakLength-1
+    return output
+
+def generate_different(chainLength, alphabet):
+    """
+    Returns a string of length='chainLength' made up from random characters from 'alphabet'.
+    It's ensured that the same character cannot appear two times in a row
+    chainLength: Expects an integer > 0 to determine the length of the string output
+    alphabet: A Python list containing all valid characters
+    """
+    output = ""
+    prevChar = ""
+    for _ in range(chainLength):
+        selectedChar = random.choice(alphabet)
+        while selectedChar==prevChar:
+            selectedChar = random.choice(alphabet)
+        output += selectedChar
+        prevChar = selectedChar
+    return output
+
+def generate_fibonacci(chainLength, alphabet):
+    match chainLength:
+        case 0:
+            return ""
+        case 1:
+            return random.choice(alphabet)
+
+    a = random.choice(alphabet)
+    b = random.choice(alphabet)
+    while a==b:
+        b = random.choice(alphabet)
+
+    while len(b)<chainLength:
+        a, b = b, a + b
+    return b[:chainLength]
+
+def randomRange(min, max):
+    if min >= max:
+        return min
+    else:
+        return random.choice(range(min,max))
+
+def generate_chains(nChains, alphabet, method, minLength, maxLength):
+    chains = []
+    
+    for i in range(nChains):
+        length = randomRange(minLength, maxLength)
+        chains.append(method(length, alphabet))
+    
+    return chains
+
+def border_array(x: str) -> list[int]:
+    """
+    Construct the border array for x.
+
+    >>> border_array("aaba")
+    [0, 1, 0, 1]
+    >>> border_array("ississippi")
+    [0, 0, 0, 1, 2, 3, 4, 0, 0, 1]
+    >>> border_array("")
+    []
+    >>> strict_border_array("abaabaa")
+    [0, 0, 1, 1, 2, 3, 4]
+    >>> border_array("abcabdabcabc")
+    [0, 0, 0, 1, 2, 0, 1, 2, 3, 4, 5, 3]
+    """
+    if x=="":
+        return []
+    border_list = [0 for _ in x]
+    for i, c in enumerate(x):
+        border_index = i
```

Comment:
Can't you just use `i` instead of `border_index`? You can count it down in the inner loop, and the outer loop will reset it to the next index as part of the `for` loop.